File: scala/src/main/scala/Consumer.scala
package ov_chipcard


object Consumer extends App {
  val log = org.slf4j.LoggerFactory.getLogger("Consumer")

  val channel = new Channel(
    host = "localhost",
    exchange = "",
    queue = "ov-chipcard"
  )

  channel subscribe { event =>
    /* Replace with consumer logic */
    log.info(s"Received event:\n$event")
  }

  log.info("Waiting for messages. To exit press ENTER")

  scala.io.StdIn.readLine()

  channel.close()
}

File: java/src/main/java/Consumer.java
package ov_chipcard;


@lombok.extern.slf4j.Slf4j(topic="Consumer")
class Consumer {

  static void handle(Event event) {
    log.info("Received event:\n{}", event.toString());

    /* Extend with actual consumer logic*/
  }


  /** Plumbing code */
  public static void main(String[] args) {
    Channel channel = new Channel();
    log.info("Waiting for messages. To exit press ENTER");
    channel.subscribe(Consumer::handle);
    System.console().readLine();
    channel.close();
  }
}

